### ISSUE TITLE
Failure: unlink(/vagrant/app/storage/3241252d6b169d30e6.html): No such file or directory

### ISSUE THREAD

**Issue by BnSmth** [cangelis/l4pdf#6]
After spending a long time trying to get wkhtmltopdf to install correctly on my vagrant vm, I am now getting the above error when using:

PDF::loadUrl('http://www.laravel.com')->download('laravel.pdf');

It's creating the pdf of the page with the same file name and this is located in the /vagrant/app/storage/ directory.

Aside from this problem I really like the package, great work.

**Comment by cangelis** [comment]
I just published an update for the [pdf generator](https://github.com/cangelis/php-pdf). Can you run `composer update` and see whether the issue is resolved or not?

**Comment by BnSmth** [comment]
I have run composer update.Using download('laravel.pdf') as I was before no longer produces an error but it is also no longer producing a file in the app/storage directory and when I navigate to the url in my browser produces a white page with no pdf being downloaded.

If I run PDF::loadUrl('http://www.laravel.com')->save('/vagrant/app/storage/laravel.pdf'); then everything is working correctly, with the pdf being created and saved in the storage directory.

Thanks for the quick response :)

**Comment by cangelis** [comment]
`app/storage` folder is used for saving temporary files so you generally don't see any file related to l4pdf because they are deleted after the pdf is generated.

I have no idea why `download()` is not working. Are you sure you are `return`ing the `download()` as a response ?

```
Route::get('/', function()
{
    return PDF::loadUrl('http://www.laravel.com')->download('test.pdf');
});
```

**Comment by BnSmth** [comment]
Aha... I was trying to get stream working and it wasn't until I realised I wasn't returning the result. Using:

return PDF::loadUrl('http://www.laravel.com')->serve('laravel.pdf');

was working as expected so I tried using download whilst returning the result and it is also working.

Having reviewed the examples on the main repo page I see I should have been doing this from the very start, I think what threw me was that the first example doesn't use return though I guess this should have been obvious.

Oh well, glad I got it solved and thank you for the rapid response and just in general for this excellent package, looking forward to using it in production.
